tests/strings/dynamic_string_check.cpp

```diff
@@ -0,0 +1,111 @@
+#include "gtest/gtest.h"
+#include "strings/DynamicString.h"
+#include <string.h>
+
+TEST(dynamic_string_tests, constructor_tests) {
+    DynamicString string1;
+    DynamicString string2("HELLo WORld!");
+    DynamicString string3(string1);
+
+    string3.empty();

```

Comment:
Try writing a test case on whether it was empty and whether it got cleared

Reply:
True, I'll add that